ibis/expr/types/relations.py

```diff
@@ -2859,6 +2859,98 @@ def info(self) -> Table:
+        │ flipper_length_mm │ int64   │   342 │ NULL   │    NULL │   172.0 │ … │
+        │ body_mass_g       │ int64   │   342 │ NULL   │    NULL │  2700.0 │ … │
+        │ sex               │ string  │   333 │ male   │     2.0 │    NULL │ … │
+        │ year              │ int64   │   344 │ NULL   │    NULL │  2007.0 │ … │
+        └───────────────────┴─────────┴───────┴────────┴─────────┴─────────┴───┘
+        >>> p.select(s.of_type("numeric")).describe()
+        ┏━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━┳━━━━━━━┳━━━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━━━━━━┳━━━┓
+        ┃ name              ┃ type    ┃ count ┃ min     ┃ mean        ┃ std        ┃ … ┃
+        ┡━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━╇━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━━━━╇━━━━━━━━━━━━╇━━━┩
+        │ string            │ string  │ int64 │ float64 │ float64     │ float64    │ … │
+        ├───────────────────┼─────────┼───────┼─────────┼─────────────┼────────────┼───┤
+        │ bill_length_mm    │ float64 │   342 │    32.1 │   43.921930 │   5.459584 │ … │
+        │ bill_depth_mm     │ float64 │   342 │    13.1 │   17.151170 │   1.974793 │ … │
+        │ flipper_length_mm │ int64   │   342 │   172.0 │  200.915205 │  14.061714 │ … │
+        │ body_mass_g       │ int64   │   342 │  2700.0 │ 4201.754386 │ 801.954536 │ … │
+        │ year              │ int64   │   344 │  2007.0 │ 2008.029070 │   0.818356 │ … │
+        └───────────────────┴─────────┴───────┴─────────┴─────────────┴────────────┴───┘
+        >>> p.select(s.of_type("string")).describe()
+        ┏━━━━━━━━━┳━━━━━━━━┳━━━━━━━┳━━━━━━━━┳━━━━━━━━┓
+        ┃ name    ┃ type   ┃ count ┃ mode   ┃ unique ┃
+        ┡━━━━━━━━━╇━━━━━━━━╇━━━━━━━╇━━━━━━━━╇━━━━━━━━┩
+        │ string  │ string │ int64 │ string │ int64  │
+        ├─────────┼────────┼───────┼────────┼────────┤
+        │ species │ string │   344 │ Adelie │      3 │
+        │ island  │ string │   344 │ Biscoe │      3 │
+        │ sex     │ string │   333 │ male   │      2 │
+        └─────────┴────────┴───────┴────────┴────────┘
+        """
+
+        quantile = sorted(quantile)
+        data = []
+        for colname in self.columns:
+            col = self[colname]
+            typ = col.type()
+            one = {}
+            one["name"] = colname
+            one["type"] = str(typ)
+            one["count"] = col.count().execute()
+            if typ.is_numeric():
+                one["min"] = col.min().execute()
+                one["mean"] = col.mean().execute()
+                one["std"] = col.std().execute()
+                for q in quantile:
+                    pname = (
+                        f"p{int(q * 100)}" if q * 100 == int(q * 100) else f"p{q * 100}"
+                    )
+                    one[pname] = col.quantile(q).execute()
+                one["max"] = col.max().execute()
+            elif typ.is_string():
+                one["mode"] = col.mode().execute()
+                one["unique"] = col.nunique().execute()
```

Comment:
All of these execute calls seem like the wrong approach. This will incur a lot of overhead in comms between the database and the client.

Is there an Ibis expression you can construct (perhaps using `ibis.union`?) that would allow you to compute the results you're after using a single query?

Reply:
Thanks, sounds great.